src/cephadm/cephadm

```diff
@@ -3548,6 +3549,33 @@ def install_sysctl(ctx: CephadmContext, fsid: str, daemon_type: str) -> None:
         call_throws(ctx, ['sysctl', '--system'])
 
 
+def sysctl_get(ctx: CephadmContext, variable: str) -> Union[str, None]:
+    """
+    Read a sysctl setting by executing 'sysctl -b {variable}'
+    """
+    out, err, code = call(ctx, ['sysctl', '-b', variable])
+    return out or None
+
+
+def filter_sysctl_settings(ctx: CephadmContext, lines: List[str]) -> List[str]:
+    """
+    Given a list of sysctl settings, examine the system's current configuration
+    and return those which are not currently set as described.
+    """
+    def testSetting(desiredLine: str) -> bool:
+        # Remove any comments
+        commentStart = desiredLine.find('#')
+        if commentStart != -1:
+            desiredLine = desiredLine[:commentStart]
+        desiredLine = desiredLine.strip()
+        if not desiredLine:
+            return False
+        setting, desiredValue = map(lambda s: s.strip(), desiredLine.split('='))
+        actualValue = sysctl_get(ctx, setting)
+        return desiredValue != actualValue
+    return list(filter(testSetting, lines))
```

Comment:
consider using python3 style: `[testSetting(line) for line in lines]`

Reply:
that would yield a list of booleans

Reply:
yeah, you are right, actually it should be: [l for l in lines if testSetting(l)]